contracts/rust/src/lib.rs

```diff
@@ -104,6 +111,23 @@ impl NEP4 for NonFungibleTokenBasic {
 
     fn transfer(&mut self, new_owner_id: AccountId, token_id: TokenId) {
         let token_owner_account_id = self.get_token_owner(token_id);
+        let signer = env::predecessor_account_id();
```

Comment:
I think we should rename the variable to `predecessor` since `signer` and `predecessor` are two different things

Reply:
Good catch! I did another pass at s/signer/predecessor everywhere